### ISSUE TITLE
Simplify The Build 

### ISSUE THREAD

**Issue by CodaFi** [typelift/SwiftCheck#243]
As of right now, SwiftCheck builds with

- Carthage
- CocoaPods
- Swift Package Manager

And maintains a linear number of build scripts and package definitions for each.

**This Is Absurd** 

We need to commit to just SwiftPM now that we're building against Swift 4.0.  My understanding is that a PodSpec can still be provided with the repo even with that kind of change, so this means we're just dropping Carthage on the floor.

**Comment by CodaFi** [comment]
This means we can eliminate the Xcodeproj as well and just ship the workspace.

**Comment by felix91gr** [comment]
SPM ftw!

I’ve never used Carthage nor CocoaPods, though. Do they even work on Linux?

Also, what did (or do) they have that SPM didn’t (or doesn’t), s.t. they were (are) attractive still?

**Comment by CodaFi** [comment]
> I’ve never used Carthage nor CocoaPods, though. Do they even work on Linux?

Now AFAIK.

> what did (or do) they have that SPM didn’t (or doesn’t), s.t. they were (are) attractive still?

Swift PM was the latecomer for our build system, which was originally based on Carthage because we had a weird dependency structure.  In addition, SwiftPM didn't support custom target layouts and when we first started using it iOS support sucked,  I eliminated that structure a while ago.  Now that the package manager has matured (in particular, so it now has test-only dependencies), we can describe our entire build with it.

**Comment by CodaFi** [comment]
The biggest tradeoff here is that we are dropping explicit builds for iOS and tvOS (watchOS still doesn't support testing targets).  I suspect, tho, that we can still recover this with `swift package generate-xcodeproj` and some `plutil` hackery to get an iOS build.

**Comment by felix91gr** [comment]
Ohh, I see. Nice :)

**Comment by akashivskyy** [comment]
Please don't drop Carthage support. 😥

**Comment by CodaFi** [comment]
You can add the framework as a submodule tracked by carthage, it just won't build a framework automatically.  You'll need to generate the Xcode project with `swift package generate-xcodeproj` then add it as a sub-project.

**Comment by broomburgo** [comment]
I agree that the state of the build systems is messy, and if the goal is keeping only one, I'd too go with Swift PM. Unfortunately dropping Carthage support (for SwiftCheck and in prospect for all the typelift libraries) is going to be extremely painful for iOS developers: Carthage was literally a godsend for iOS devs that didn't want to lose control of their project to CocoaPods, and has become basically a given for many.

**Comment by CodaFi** [comment]
We won't be dropping iOS support, we'll just be dropping Carthage's iOS support.  I don't intend to do this now, but when and if we come up with a satisfactory solution that keeps our per-platform builds working - or if Apple decides to allow multiple platforms in one package def - we will be switching
